src/auth/core/token.decorator.ts

```diff
@@ -0,0 +1,9 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { JwtTokenPayload } from './jwt-auth.guard';
+
+export const Token = createParamDecorator(
```

Comment:
Is this used? Didn't see it in the code

Reply:
Will be used soon to check the user id in the token.

Reply:
```suggestion
/**
 * Extract token payload for advanced checks.
 */
export const Token = createParamDecorator(
```